File: qabelbox/src/test/java/de/qabel/qabelbox/storage/server/BlockServerTest.java
package de.qabel.qabelbox.storage.server;

import android.support.annotation.Nullable;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricGradleTestRunner;
import org.robolectric.RuntimeEnvironment;
import org.robolectric.annotation.Config;

import de.qabel.qabelbox.BuildConfig;
import de.qabel.qabelbox.SimpleApplication;
import de.qabel.qabelbox.communication.callbacks.JSONModelCallback;
import de.qabel.qabelbox.config.AppPreference;
import de.qabel.qabelbox.storage.data.BoxQuotaJSONAdapter;
import de.qabel.qabelbox.storage.model.BoxQuota;
import de.qabel.qabelbox.util.TestHelper;
import okhttp3.Response;

import static junit.framework.Assert.assertEquals;

@RunWith(RobolectricGradleTestRunner.class)
@Config(application = SimpleApplication.class, constants = BuildConfig.class)
public class BlockServerTest {

    private BlockServer blockServer;

    @Before
    public void setUp() {
        blockServer = new MockBlockServer();
    }

    @Test
    public void testQuota() throws Exception {
        BoxQuota[] quota = new BoxQuota[1];
        blockServer.getQuota(new JSONModelCallback<BoxQuota>(new BoxQuotaJSONAdapter()) {
            @Override
            protected void onSuccess(Response response, BoxQuota model) {
                quota[0] = model;
            }

            @Override
            protected void onError(Exception e, @Nullable Response response) {
                //Should not happen
            }
        });

        BoxQuota boxQuota = quota[0];
        assertEquals(MockBlockServer.SIZE, boxQuota.getSize());
        assertEquals(MockBlockServer.QUOTA, boxQuota.getQuota());
    }

}

File: qabelbox/src/test/java/de/qabel/qabelbox/storage/BlockServerTest.kt
package de.qabel.qabelbox.storage

import de.qabel.qabelbox.BuildConfig
import de.qabel.qabelbox.SimpleApplication
import de.qabel.qabelbox.config.AppPreference
import de.qabel.qabelbox.storage.server.AndroidBlockServer
import de.qabel.qabelbox.storage.server.BlockServer
import de.qabel.qabelbox.test.TestConstants
import org.junit.Before
import org.junit.Assert.*
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricGradleTestRunner
import org.robolectric.RuntimeEnvironment
import org.robolectric.annotation.Config

@RunWith(RobolectricGradleTestRunner::class)
@Config(application = SimpleApplication::class, constants = BuildConfig::class)
class BlockServerTest() {

    lateinit var server: BlockServer

    @Before
    fun setUp() {
        server = AndroidBlockServer(AppPreference(RuntimeEnvironment.application), RuntimeEnvironment.application)
    }

    @Test
    fun testUrlForFile(){
        val uuid = "65eca15b-0ccc-46c7-9c91-a60917ff183d"
        val prefix = TestConstants.PREFIX

        val urlA = "https://block.qabel.org/api/v0/files/$prefix/$uuid"
        var url = server.urlForFile(TestConstants.PREFIX, urlA)
        assertEquals(urlA, url)

        url = server.urlForFile(TestConstants.PREFIX, "blocks/$uuid")
        assertEquals(url, "${TestConstants.BLOCK_URL}/api/v0/files/$prefix/blocks/$uuid")
    }


}
